docs/api/utils.md

```diff
@@ -180,6 +180,22 @@ const countReducerAtom = atomWithReducer(0, countReducer)
 
 https://codesandbox.io/s/react-typescript-forked-g3tsx
 
+## atomWithDefault
+
+Ref: https://github.com/pmndrs/jotai/issues/352
+
+This is a function to create a primitive atom.
+Its default value can be specified with a read function instead of a static initial value.
+
+```js
+import { atomWithDefault } from 'jotai/utils'
+
+const count1Atom = atom(1)
+const count2Atom = atomWithDefault((get) => get(count1Atom) * 2)
```

Comment:
Question:
Is it possible to add a variant, where you can add a default value in atom? Could be useful, if you want to base the default value on props.

Reply:
I'm not sure if I get "add a default value in atom".
Could you write some rough idea/code?

Reply:
```
const DogInfo: FC = (props) => {
  const dogInfo = useAtom(dogAtom(props.dogName))
  // ...
}
```

Reply:
Ah, I want to support this pattern with this:
```js
const DogInfo = ({ dogName }) => {
  const dogAtom = useMemo(() => atom(dogName), [dogName])
  const dogInfo = useAtom(dogAtom)
  // ...
}
```

Oh, I see you want to give the default that can be overwritten.
```js
const DogInfo = ({ dogName }) => {
  const dogAtom = useRef(atom(dogName)).current
  const dogInfo = useAtom(dogAtom)
  // ...
}
```
I'm not a big fan of this useRef pattern.

```js
const DogInfo = ({ dogName }) => {
  const dogAtom = useMemo(() => atom(null), [])
  const dogInfo = useAtom(dogAtom) || dogName
  // ...
}
```

There might be some better ideas. I feel like using atomWithDefault for props is a misuse.